src/Node/CaptureNode.php

```diff
@@ -27,62 +28,25 @@ public function __construct(Node $body, int $lineno, ?string $tag = null)
 
     public function compile(Compiler $compiler): void
     {
-        if ($compiler->getEnvironment()->useYield()) {
-            if ($this->getAttribute('raw')) {
-                $compiler->raw("implode('', iterator_to_array(");
-            } else {
-                $compiler->raw("('' === \$tmp = implode('', iterator_to_array(");
-            }
-            if ($this->getAttribute('with_blocks')) {
-                $compiler->raw("(function () use (&\$context, \$macros, \$blocks) {\n");
-            } else {
-                $compiler->raw("(function () use (&\$context, \$macros) {\n");
-            }
-            $compiler
-                ->indent()
-                ->subcompile($this->getNode('body'))
-                ->outdent()
-                ->write("})() ?? new \EmptyIterator()))")
-            ;
-            if (!$this->getAttribute('raw')) {
-                $compiler->raw(") ? '' : new Markup(\$tmp, \$this->env->getCharset())");
-            }
-            $compiler->raw(';');
-
-            return;
+        if ($this->getAttribute('raw')) {
+            $compiler->raw("implode('', iterator_to_array(");
+        } else {
+            $compiler->raw("('' === \$tmp = implode('', iterator_to_array(");
         }
-
         if ($this->getAttribute('with_blocks')) {
             $compiler->raw("(function () use (&\$context, \$macros, \$blocks) {\n");
         } else {
             $compiler->raw("(function () use (&\$context, \$macros) {\n");
         }
-        $compiler->indent();
-        if ($compiler->getEnvironment()->isDebug()) {
-            $compiler->write("ob_start();\n");
-        } else {
-            $compiler->write("ob_start(function () { return ''; });\n");
```

Comment:
When not in yield-only mode, don't we need to handle the output buffer in this capture node to ensure we are properly capturing everything including legacy output nodes ?

Reply:
indeed, updated